lib/render.ml

```diff
@@ -13,13 +15,14 @@ let scale_point size point =
   (x, y)
 
 let draw_circle ctx ({ c; radius } : circle) =
-  let x, y = scale_point ctx.size c in
-  let radius = radius /. min (fst ctx.size) (snd ctx.size) in
+  let size = tmap float_of_int ctx.size in
+  let x, y = scale_point size c in
+  let radius = radius /. min (fst size) (snd size) in
```

Comment:
Why do we change the radius like this? 

I don't think this would give us what we want on a rectangular canvas.

Reply:
It has to be scaled down to a 0-1 range some way, I'm not sure what a better alternative is.

Reply:
I'm currently co-working with a friend who is a professional graphics programmer, she thinks it should be this `radius /. (sqrt (Float.pow width 2. +. Float.pow height 2.) /. 2.) `

Reply:
It depends on how we convert one unit of joy-land into one unit of cairo.
If you look at the new ticket I created - https://github.com/Sudha247/ocaml-joy/issues/86 - this is caused because we are converting a `(float_in_range[-150..150], float_in_range[-250..250])` to `(float_in_range[0..1], float_in_range[0..1])`. It wouldn't be right to map a `rectangle` canvas to a `square` canvas. We should map to a rectangular canvas.

Reply:
I learnt that we don't need to map it to `[0..1]` either.
https://github.com/Chris00/ocaml-cairo/blob/1a347fb5d7cc887121bc335ecd3dfc68a79f0318/examples/stroke.ml#L6-L18

We can denormalize points to `([0..w], [0..h])` and that would be okay if we omit `Cairo.scale w h` later.

Reply:
I'm not sure I understand how that example shows that, the coordinates are all in [0, 1]

Reply:
It's scaled [0, 1] on both axes, with respect to that axis so no information should be lost so to speak. I've switched to the Euclidean norm for the radius and running the circle examples with a rectangular aspect ratio, I am not getting the distortions you're getting.

Reply:
The coordinates are scaled in [0..1] with respect to that axis. So scaling a Joy coordinate space of size [200, 300] to [0..1] on both axes does not scale it into a square aspect ratio. No resolution is 'lost', so to speak, as those [0..1] ranges mean different things for each axis. 

This code also just gives us a static radius for any given circle, I feel confident that it isn't what's causing the distortions we're seeing in non-square aspect ratios.